### ISSUE TITLE
restic crashes when backup location runs out of space

### ISSUE THREAD

**Issue by morphis** [restic/restic#1217]
## Output of `restic version`

restic compiled manually
compiled with go1.9 on linux/amd64

Based on git revision b46774be

## How did you run restic exactly?

```
/snap/bin/restic -r REPO="sftp:user@server:/path/to/backup/dir backup \
    --tag home \
    $HOME
```

## What backend/server/service did you use?

sftp

## Expected behavior

Restic should exit with a error message that either space on the remote backup location is not available or that something while transferring data over SFTP has failed.

## Actual behavior

restic crashes:

```
scanned 7040 directories, 49817 files in 0:01
panic: Write: sftp: "Failure" (SSH_FX_FAILURE)08 GiB  56850 / 56857 items  0 errors  ETA 6:46 

goroutine 79105 [running]:
github.com/restic/restic/internal/archiver.(*Archiver).saveChunk(0xc42020d810, 0xe09320, 0xc420016190, 0xf1dfee8c, 0xb19e5, 0xb108f0ac00000, 0xc42e256000, 0xb19e5, 0x75e000, 0xc4203a7ad0, ...)
	/home/simon/Work/Personal/restic/parts/restic/go/src/github.com/restic/restic/internal/archiver/archiver.go:165 +0x39d
created by github.com/restic/restic/internal/archiver.(*Archiver).SaveFile
	/home/simon/Work/Personal/restic/parts/restic/go/src/github.com/restic/restic/internal/archiver/archiver.go:239 +0x3ff
```

## Steps to reproduce the behavior

1. Ensure remote backup location has less space than required for a new backup
2. Run restic as described under "How did you run restic exactly?"

## Do you have any idea what may have caused this?

I didn't looked into the code yet but it looks like restic isn't correctly handling errors returned from the SFTP file transfer.

**Comment by fd0** [comment]
Thanks for raising this issue. We'll see what we can do.

**Comment by martinseener** [comment]
As far as i know, the SFTP commands do not offer such a functionality but you could try to use `df` which works on linux-based boxes (not tested BSD, macOS or windows though) and use this, if available to give hint

**Comment by MichaelEischer** [comment]
Closing this in favor of #1153. The crash itself has been fixed sometime between 0.7.1 and 0.9.0.
